libcalico-go/lib/validator/v3/validator.go

```diff
@@ -1560,18 +1564,54 @@ func validateRuleMetadata(structLevel validator.StructLevel) {
 
 func validateRouteTableRange(structLevel validator.StructLevel) {
 	r := structLevel.Current().Interface().(api.RouteTableRange)
-	if r.Min >= 1 && r.Max >= r.Min && r.Max <= 250 {
-		log.Debugf("RouteTableRange is valid: %v", r)
-	} else {
+
+	if r.Min > r.Max {
+		log.Warningf("RouteTableRange is invalid: %v", r)
+		structLevel.ReportError(
+			reflect.ValueOf(r),
+			"RouteTableRange",
+			"",
+			reason("min value cannot be greater than max value"),
+			"",
+		)
+	}
+
+	if r.Min <= 0 {
+		log.Warningf("RouteTableRange is invalid: %v", r)
+		structLevel.ReportError(
+			reflect.ValueOf(r),
+			"RouteTableRange",
+			"",
+			reason("cannot target indices < 1"),
+			"",
+		)
+	}
+
+	// cast both ints to 64bit as casting the max 32-bit integer to int() would overflow on 32bit systems
+	if int64(r.Max) > int64(routeTableMaxLinux) {
 		log.Warningf("RouteTableRange is invalid: %v", r)
 		structLevel.ReportError(
 			reflect.ValueOf(r),
 			"RouteTableRange",
 			"",
-			reason("must be a range of route table indices within 1..250"),
+			reason("max index too high"),
 			"",
 		)
 	}
+
+	// check if ranges collide with reserved linux tables
+	for _, rsrv := range routeTablesReservedLinux {
+		if r.Min == rsrv || (r.Min < rsrv && r.Max >= rsrv) {
```

Comment:
Why not use the `<=` operator?

Reply:
Had a flu all last week, going to blame that 😆